config/initializers/warden.rb

```diff
@@ -6,19 +6,24 @@
 require 'open_project/authentication/strategies/warden/user_basic_auth'
 require 'open_project/authentication/strategies/warden/session'
 
-strategies = {
-  basic_auth_failure: OpenProject::Authentication::Strategies::Warden::BasicAuthFailure,
-  global_basic_auth:  OpenProject::Authentication::Strategies::Warden::GlobalBasicAuth,
-  user_basic_auth:    OpenProject::Authentication::Strategies::Warden::UserBasicAuth,
-  session:            OpenProject::Authentication::Strategies::Warden::Session
-}
+WS = OpenProject::Authentication::Strategies::Warden
+
+strategies = [
+  [:basic_auth_failure, WS::BasicAuthFailure, 'Basic'],
+  [:global_basic_auth,  WS::GlobalBasicAuth,  'Basic'],
+  [:user_basic_auth,    WS::UserBasicAuth,    'Basic'],
```

Comment:
The way to go here would probably be to modify all requests made with `$http` by manipulating the default headers (or introducing an interceptor).

This can be done [here](https://github.com/opf/openproject/blob/release/4.2/frontend/app/openproject-app.js#L196)